presto-spark-base/src/main/java/com/facebook/presto/spark/execution/http/PageResponseHandler.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution.http;
+
+import com.facebook.airlift.http.client.HttpStatus;
+import com.facebook.airlift.http.client.Request;
+import com.facebook.airlift.http.client.Response;
+import com.facebook.airlift.http.client.ResponseHandler;
+import com.facebook.presto.operator.PageTransportErrorException;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.spi.page.SerializedPage;
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.MediaType;
+import io.airlift.slice.InputStreamSliceInput;
+import io.airlift.slice.SliceInput;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
+import static com.facebook.airlift.http.client.ResponseHandlerUtils.propagate;
+import static com.facebook.presto.PrestoMediaTypes.PRESTO_PAGES_TYPE;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_BUFFER_COMPLETE;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_PAGE_NEXT_TOKEN;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_PAGE_TOKEN;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_TASK_INSTANCE_ID;
+import static com.facebook.presto.spi.page.PagesSerdeUtil.readSerializedPages;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class PageResponseHandler
+        implements ResponseHandler<PagesResponse, RuntimeException>
```

Comment:
same comments as the other one; could add comment/javadoc to explain why do we need to pull shuffle data from native executor?